src/components/subscriptions/buttons/DownloadCsvButton.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useContext, useState } from 'react';
+import { CSVLink } from 'react-csv';
+import { logError } from '@edx/frontend-platform/logging';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { SubscriptionDetailContext } from '../SubscriptionDetailContextProvider';
+import LicenseManagerApiService from '../data/service';
+import { ALL_USERS } from '../data/constants';
+import { configuration } from '../../../config';
+
+const DownloadCsvButton = ({ enterpriseSlug }) => {
+  const { subscription } = useContext(SubscriptionDetailContext);
+  const [csvData, setCsvData] = useState([]);
+
+  const getCsvFileName = () => {
+    const titleNoWhitespace = subscription.title.replace(/\s+/g, '');
+    const currentDate = new Date();
+    const year = currentDate.getUTCFullYear();
+    const month = currentDate.getUTCMonth() + 1;
+    const day = currentDate.getUTCDate();
+    return `${titleNoWhitespace}-${year}-${month}-${day}.csv`;
+  };
+
+  const getLicenseActivationLink = (activationKey) => (
+    `${configuration.ENTERPRISE_LEARNER_PORTAL_URL}/${enterpriseSlug}/licenses/${activationKey}/activate`
+  );
+
+  const options = {
+    status: ALL_USERS,
+  };
+
+  const headers = [
+    { label: 'Status', key: 'status' },
+    { label: 'User Email', key: 'userEmail' },
+    { label: 'Activation Date', key: 'activationDate' },
+    { label: 'Last Remind Date', key: 'lastRemindDate' },
+    { label: 'License Activation Link', key: 'activationLink' },
+  ];
+
+  LicenseManagerApiService.fetchSubscriptionUsers(subscription.uuid, options, 5000)
```

Comment:
A few thoughts:
1. What happens if the subscriptions data has more than 5000 records? As is, this API call would only return the first 5000 in the downloaded CSV file. We get around this in other places in frontend-app-admin-portal by disabling pagination on the API call to fetch the CSV data (to get all pages) and have the UI download that response as a file. On the API side, this is done using the `rest_framework_csv.renderers.CSVRenderer` extension for django-rest-framework, which allows you to add `.csv` to an endpoint for it to return the data formatted as a CSV.
* https://github.com/mjumbewu/django-rest-framework-csv
* Example of disabling pagination on API request to fetch full CSV report: https://prod-edx-analyticsapi.edx.org/enterprise/api/v0/enterprise/8ea70d02-c210-4009-a86d-e23423dcda7d/enrollments.csv/?no_page=true
2. Do we need to fetch this data every time the component re-renders? Would it make more sense if the API call is made when the user clicks the download button (where the download button is a [StatefulButton](https://paragon-edx.netlify.app/components/statefulbutton/)). Even if we did fetch the data prior to button click, we'd only need to fetch the data once by using an `useEffect` hook.
3. There is an existing `DownloadCsvButton` component in this MFE that accepts a few props like a fetch method; was it possible to use/extend that existing component vs. creating a separate component? Either way, that existing component could be cleaned up / refactored to no longer rely on Redux and use `StatefulButton`, too, so I don't feel strongly that we absolutely _need_ to use the existing `DownloadCsvButton` component here.

Reply:
Button is now implemented using the `StatefulButton` component from Paragon and only fetches data on button click. The new license CSV endpoint in license-manager is now used to fetch the data in CSV format. PR for CSV endpoint: https://github.com/edx/license-manager/pull/200